### DISCUSSION TITLE
Silent/background updates not working

### DISCUSSION THREAD

**Discussion by nishant-nf** [sparkle-project/Sparkle#2566] [Q&A] [answered]
I am trying to have sparkle automatically download an install the update without any user intervention. I have set the following value in Info.plist
```
    <key>SUEnableAutomaticChecks</key>
    <true/>
    <key>SUScheduledCheckInterval</key>
    <integer>3600</integer>
    <key>SUAutomaticallyUpdate</key>
    <true/>
    <key>SUAllowsAutomaticUpdates</key>
    <true/>
```
The seed URL and the public key are set too.
However, after the hour for the update period elapses, I see the popup showing that a new version is available. 
![Screenshot 2024-05-21 at 16 46 57](https://github.com/sparkle-project/Sparkle/assets/119282733/db65fc64-5c27-4dc9-9fcb-0356c65f8a5c)

Is there any additional configuration/code required to make this work? Based on the documentation it didn't seem like there was.
The app is in the Applications folder and Finder does not request additional permissions if I try to move the app to the Trash. The update is also just a zip of the .app.

**Comment by zorgiepoo** [comment]
Can you print out the value of [automaticallyDownloadsUpdates](https://sparkle-project.github.io/documentation/api-reference/Classes/SPUUpdater.html#/c:objc(cs)SPUUpdater(py)automaticallyDownloadsUpdates) on your `SPUUpdater` and see if it is NO?

Better yet, run:
```
defaults read your-app-bundle-id SUAutomaticallyUpdate
```

[SUAutomaticallyUpdate](https://sparkle-project.org/documentation/customization/) specifies the initial value for automatically downloading updates, but if the user in your app says "I don't want to automatically download updates" (like from Sparkle's permission prompt on a prior run or from the software update window) or you programmatically set the value for automaticallyDownloadsUpdates in your app before, that will take precedence (hence automaticallyDownloadsUpdates should only be set in response to user setting changes typically).

**Reply by nishant-nf** [reply]
Here's the output
```
% defaults read ai.nightfall.MacLandmarks SUAutomaticallyUpdate

1
% defaults read-type ai.nightfall.MacLandmarks SUAutomaticallyUpdate

Type is boolean
```

**Reply by zorgiepoo** [reply, marked as answer]
What if you enable the checkbox in the update window? Does your app remember it then? Is your app sandboxed or was it sandboxed before? (User defaults may not migrate cleanly if a sandboxed app becomes not sandboxed)

**Reply by zorgiepoo** [reply]
Check the value of the SPUUpdater property too

**Reply by nishant-nf** [reply]
`automaticallyDownloadsUpdates` was getting printed as `true`. The app was sandboxed by default in Xcode and I had removed it before creating the current builds, but I had run it as a sandboxed one before. I created a new app that wasn't sandboxed and it did get updated in the background with these settings.
Thanks for your help.
